File: AdMobMediation_v8/ViewController/Banner/BannerViewController.m
//
//  BannerViewController.m
//  AdMobMediation_v8
//
//  Created by JustinTsou on 2021/5/31.
//

#import "BannerViewController.h"
#import <GoogleMobileAds/GoogleMobileAds.h>

static NSString *const BannerAdUnit = @"You banner ad unit";

@interface BannerViewController ()<GADBannerViewDelegate> {
    GADBannerView *_gadBannerView;
}

@property (weak, nonatomic) IBOutlet UIButton *refreshButton;
@end

@implementation BannerViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view from its nib.
    
    [self setupGADBannerView];
}

- (IBAction)tapRefreshButton:(UIButton *)sender {
    [_gadBannerView loadRequest:[GADRequest request]];
}

#pragma mark : Setup GADBanner

- (void)setupGADBannerView {
    _gadBannerView = [[GADBannerView alloc]initWithAdSize:kGADAdSizeBanner];
    _gadBannerView.delegate = self;
    _gadBannerView.rootViewController = self;
    _gadBannerView.adUnitID = BannerAdUnit;
    [_gadBannerView loadRequest:[GADRequest request]];
}

- (void)setupBannerViewUI:(GADBannerView *)bannerView {
    
    [self.view addSubview:bannerView];

    [bannerView setTranslatesAutoresizingMaskIntoConstraints:NO];

    [bannerView.heightAnchor constraintEqualToConstant:bannerView.bounds.size.height].active = YES;
    [bannerView.bottomAnchor constraintEqualToAnchor:self.view.safeAreaLayoutGuide.bottomAnchor constant:0.0].active = YES;
    [bannerView.leftAnchor constraintEqualToAnchor:self.view.leftAnchor constant:0].active = YES;
    [bannerView.rightAnchor constraintEqualToAnchor:self.view.rightAnchor constant:0].active = YES;
}

#pragma mark :GADBannerViewDelegate

- (void)bannerViewDidReceiveAd:(GADBannerView *)bannerView {
    if (bannerView != nil) {
        [self setupBannerViewUI:bannerView];
    }
}

- (void)bannerView:(GADBannerView *)bannerView didFailToReceiveAdWithError:(NSError *)error {
    NSLog(@"error message: %@",error.localizedDescription);
}

@end

File: AdMobMediation_v8_Swift/ViewController/Banner/BannerViewController.swift
//
//  BannerViewController.swift
//  AdMobMediation_v8_Swift
//
//  Created by JustinTsou on 2021/5/31.
//

import UIKit
import GoogleMobileAds

class BannerViewController: UIViewController {
    
    @IBOutlet weak var RefreshButton: UIButton!
    
    private let TestBannerAdUnit: String = "Your banner ad unit"
    
    private var gadBannerView: GADBannerView?
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view.
        
        self.setupGADBannerView()
    }
    
    @IBAction func tapRefreshBtn(_ sender: UIButton) {
        
        self.gadBannerViewLoadRequest()
    }
}

extension BannerViewController {
    
    private func setupGADBannerView() {
        
        gadBannerView = GADBannerView(adSize: kGADAdSizeBanner)
        gadBannerView?.delegate = self
        gadBannerView?.rootViewController = self
        gadBannerView?.adUnitID = TestBannerAdUnit
        
        self.gadBannerViewLoadRequest()
    }
    
    private func gadBannerViewLoadRequest() {
        gadBannerView?.load(GADRequest())
    }
}

extension BannerViewController: GADBannerViewDelegate {
    
    func bannerViewDidReceiveAd(_ bannerView: GADBannerView) {
        
        self.view.addSubview(bannerView)

        bannerView.translatesAutoresizingMaskIntoConstraints = false
        bannerView.heightAnchor.constraint(equalToConstant: 50).isActive = true
        bannerView.bottomAnchor.constraint(equalTo: self.view.safeAreaLayoutGuide.bottomAnchor).isActive = true
        bannerView.trailingAnchor.constraint(equalTo: self.view.trailingAnchor).isActive = true
        bannerView.leadingAnchor.constraint(equalTo: self.view.leadingAnchor).isActive = true
    }

    func bannerView(_ bannerView: GADBannerView, didFailToReceiveAdWithError error: Error) {
        print("error message:\(error.localizedDescription)")
    }
}

